examples/AnimalAndFlower/model.py

```diff
@@ -0,0 +1,180 @@
+        x = x.view(-1, output_size)
+        return x
+
+
+def model_declaration():
+    """
+    this function creates and defines the structure of our graph model
+
+    @return: a program based on the graph and it solver
+    """
+    graph.detach()
+    image = graph['image']
+    animal = graph['animal']
+    cat = graph['cat']
+    dog = graph['dog']
+    monkey = graph['monkey']
+    squirrel = graph['squirrel']
+
+    flower = graph['flower']
+    daisy = graph['daisy']
+    dandelion = graph['dandelion']
+    rose = graph['rose']
+    sunflower = graph['sunflower']
+    tulip = graph['tulip']
+
+    image['pixels'] = ReaderSensor(keyword='pixels', device=device)
+    image[animal] = ReaderSensor(keyword='animal', label=True, device=device)
+    image[cat] = ReaderSensor(keyword='cat', label=True, device=device)
+    image[dog] = ReaderSensor(keyword='dog', label=True, device=device)
+    image[monkey] = ReaderSensor(keyword='monkey', label=True, device=device)
+    image[squirrel] = ReaderSensor(keyword='squirrel', label=True, device=device)
+    image[flower] = ReaderSensor(keyword='flower', label=True, device=device)
+    image[daisy] = ReaderSensor(keyword='daisy', label=True, device=device)
+    image[dandelion] = ReaderSensor(keyword='dandelion', label=True, device=device)
+    image[rose] = ReaderSensor(keyword='rose', label=True, device=device)
+    image[sunflower] = ReaderSensor(keyword='sunflower', label=True, device=device)
+    image[tulip] = ReaderSensor(keyword='tulip', label=True, device=device)
+
+    image['emb'] = ModuleLearner('pixels', module=ImageNetwork(), device=device)
+    image[animal] = ModuleLearner('emb', module=nn.Linear(output_size, 2), device=device)
+    image[cat] = ModuleLearner('emb', module=nn.Linear(output_size, 2), device=device)
+    image[dog] = ModuleLearner('emb', module=nn.Linear(output_size, 2), device=device)
+    image[monkey] = ModuleLearner('emb', module=nn.Linear(output_size, 2), device=device)
+    image[squirrel] = ModuleLearner('emb', module=nn.Linear(output_size, 2), device=device)
+    image[flower] = ModuleLearner('emb', module=nn.Linear(output_size, 2), device=device)
+    image[daisy] = ModuleLearner('emb', module=nn.Linear(output_size, 2), device=device)
+    image[dandelion] = ModuleLearner('emb', module=nn.Linear(output_size, 2), device=device)
+    image[rose] = ModuleLearner('emb', module=nn.Linear(output_size, 2), device=device)
+    image[sunflower] = ModuleLearner('emb', module=nn.Linear(output_size, 2), device=device)
+    program = IMLProgram(graph, poi=(image,), inferTypes=['ILP', 'local/argmax'],
+                         loss=MacroAverageTracker(BCEWithLogitsIMLoss(lmbd=0.5)),
+                         metric={'ILP': PRF1Tracker(DatanodeCMMetric()),
+                                 'softmax': PRF1Tracker(DatanodeCMMetric('local/argmax'))})
+    return program
+
+
+
+def train_with_single_network(train_set, validation_set, labelKeys):
+    baseNet = ImageNetwork()
```

Comment:
Do you use this function? de we need it?

Reply:
this was meant for a test I did